src/Collections/Sequences/Seq.dfy

```diff
@@ -200,6 +200,28 @@ module Seq {
     }
   }
 
+  /* A sequence with cardinality equal to its set has no duplicates */
+  lemma LemmaNoDuplicatesCardinalityOfSet<T>(s: seq<T>)
+    requires |ToSet(s)| == |s|
+    ensures HasNoDuplicates(s)
+  {
+    reveal HasNoDuplicates();
+    reveal ToSet();
+    if |s| == 0 {
+    } else {
+      assert s == [First(s)] + DropFirst(s);
+      assert ToSet(s) == {First(s)} + ToSet(DropFirst(s));
+      if First(s) in DropFirst(s) {
+        assert ToSet(s) == ToSet(DropFirst(s));
+        LemmaCardinalityOfSet(DropFirst(s));
+        assert |ToSet(s)| <= |DropFirst(s)|;
```

Comment:
Might just be worth a comment that this implies a contradiction with the precondition. AFAICT the contrapositive technique isn't widely used in this repo so it's worth calling it out for the sake of reader understanding.

Reply:
Added the comment!